docs/userguide/dependency_management.txt

```diff
@@ -2,71 +2,144 @@
 Dependencies Management in Setuptools
 =====================================
 
-Declaring Dependencies
-======================
+``Setuptools``'s dependency management is meticulous, or agonizing, depending
+on your level of familiarity. There are three types of dependency styles.
+1) those required to run the packaging program (build system requirement),
+2) those your package depends on (required dependency) and 3) optional
+dependency.
 
-``setuptools`` supports automatically installing dependencies when a package is
-installed, and including information about dependencies in Python Eggs (so that
-package management tools like pip can use the information).
+.. Note::
+    For all the packages you intend to add to dependency, you can optionally
+    specify the version following :ref:`reference on version specifyer <WIP>`
```

Comment:
These specifiers are standardized (PEP 440) and implemented in the `packaging` package. I recommend linking to those references.